### DISCUSSION TITLE
When inlineSearchInput is true Then How would I know in my React Component its in Focus/User Typing?

### DISCUSSION THREAD

**Discussion by rajkumardev97** [dowjones/react-dropdown-tree-select#575] [Q&A]
Here is my code:
 <React.Fragment>
      <DropdownTreeSelect
        data={data}
        showDropdown={"always"}
        inlineSearchInput={true}
        texts={{
          placeholder: "Choose...",
          inlineSearchPlaceholder: "Search..."
        }}
        showPartiallySelected={true}
        onChange={onChange}
        className="mdl-dropdown-tree"
      />
    </React.Fragment>

Expecting Output: Whenever user typing on inlineSearchInput of this library How would I know he is engaging with your inlineSearchInput If you have provided any callback function that will be helpful.

  **const inlineSearchInputOnChange = () => {
    // Here Need to fire google event whenever user engage with inlineSearchInput in library.
  };**

<React.Fragment>
      <DropdownTreeSelect
        data={data}
        showDropdown={"always"}
        inlineSearchInput={true}
        texts={{
          placeholder: "Choose...",
          inlineSearchPlaceholder: "Search..."
        }}
        showPartiallySelected={true}
        onChange={onChange} 
       **anyInlineSearchInputOnChange = {inlineSearchInputOnChange }**  
        className="mdl-dropdown-tree"
      />
    </React.Fragment>

**Comment by mrchief** [comment]
There is no event to bubble up when user is typing something in the box. There is some maintenance that needs to happen like debouncing, building search tree etc. on each keystroke so raising external events would slow things down a bit. However, there is no userland interaction that we had in mind that would need such an event to be exposed. What's your use case?

**Reply by rajkumardev97** [reply]
If you can provide inFocus callback for this inline search input then that will also work for me, I just need to fire Google event from my component when user engaging with this 3rd party inline search input.

**Reply by mrchief** [reply]
I feel like there should be way either via react hooks or dom event handlers to do this already.
